python/common/org/python/types/Bytes.java

```diff
@@ -971,10 +972,56 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
     }
 
     @org.python.Method(
-            __doc__ = "B.expandtabs(tabsize=8) -> copy of B\n\nReturn a copy of B where all tab characters are expanded using spaces.\nIf tabsize is not given, a tab size of 8 characters is assumed."
+            __doc__ = "B.expandtabs(tabsize=8) -> copy of B\n\nReturn a copy of B where all tab characters are expanded using spaces.\nIf tabsize is not given, a tab size of 8 characters is assumed.",
+            default_args = {"tabsize"}
     )
-    public org.python.Object expandtabs(java.util.List<org.python.Object> args, java.util.Map<java.lang.String, org.python.Object> kwargs, java.util.List<org.python.Object> default_args, java.util.Map<java.lang.String, org.python.Object> default_kwargs) {
-        throw new org.python.exceptions.NotImplementedError("bytes.expandtabs has not been implemented.");
+    public org.python.Object expandtabs(org.python.types.Object tabsize) {
+        // CPython implementation https://github.com/python/cpython/blob/3.6/Objects/stringlib/transmogrify.h#L21
+        // TODO: check overflows
+
+        if (this.value == null) {
+            return null;
+        }
+
+        if (this.value.length == 0) {
+            return new org.python.types.Bytes(new byte[0]);
+        }
+
+        int tabsizeInt = 8;
+
+        if (tabsize != null) {
+            tabsizeInt = (int) ((org.python.types.Int) tabsize).value;
+        }
+
+        ArrayList<Character> buffer = new ArrayList<>();
+
+        for (int i = 0, lineIndex = 0; i < this.value.length; i++) {
+            char c = (char) this.value[i];
+
+            if (c == '\t') {
+                if (tabsizeInt > 0) {
+                    int untilClosestColumn = tabsizeInt - (lineIndex % tabsizeInt);
+                    lineIndex += untilClosestColumn;
+                    while (untilClosestColumn-- > 0) {
+                        buffer.add(' ');
+                    }
+                }
+            } else {
+                lineIndex++;
+                buffer.add(c);
+
+                if (c == '\n' || c == '\r') {
+                    lineIndex = 0;
+                }
+            }
+        }
+
+        byte[] bytes = new byte[buffer.size()];
+        for (int i = 0; i < buffer.size(); i++) {
+            bytes[i] = (byte) ((char) buffer.get(i));
+        }
```

Comment:
This conversion from `ArrayList<Byte>` to `byte[]` would probably be useful as a `listToArray` method, to be reused by other code.

Reply:
Where should I place that new method?

Reply:
You can start by just creating a public static method in this same class.
Another option is to create a `BytesUtils.java` file for adding static methods like these, since they are probably going to be used in `ByteArray.java` as well.